workflow_scripts/test_models.py

```diff
@@ -1,48 +1,84 @@
- cwd=cwd_path, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-stdoutput, stderroutput = obtain_diff.communicate()
-diff_list = stdoutput.split()
-
-# identify list of changed onnx models in model Zoo
-model_list = [str(model).replace("b'","").replace("'", "") for model in diff_list if ".onnx" in str(model)]
-
-# run lfs install before starting the tests
-run_lfs_install()
-
-print("\n=== Running ONNX Checker on added models ===\n")
-# run checker on each model
-failed_models = []
-for model_path in model_list:
-    model_name = model_path.split('/')[-1]
-    print("Testing:", model_name)
-
-    try:
-      pull_lfs_file(model_path)
-      model = onnx.load(model_path)
-      onnx.checker.check_model(model)
-      print("Model", model_name, "has been successfully checked!")
-    except Exception as e:
-      print(e)
-      failed_models.append(model_path)
-
-if len(failed_models) != 0:
-  print(str(len(failed_models)) +" models failed onnx checker.")
-  sys.exit(1)
-
-print(len(model_list), "model(s) checked.")
+def check_by_target(model_path, file_name, test_data_set, target):
+  if target == 'onnx' or target == 'None':
+    check_model.run_onnx_checker(model_path, file_name)
+  if target == 'onnxruntime' or target == 'None':
+    check_model.run_backend_ort(model_path, file_name, test_data_set)
+
+def main():
+  parser = argparse.ArgumentParser(description='Test settings')
+  # default None: test by both onnx and onnxruntime
+  # if target is specified, only test by the specified one
+  parser.add_argument('--target', required=False, default='None', type=str, 
+                      help='Test the model by which (onnx/onnxruntime)?',
+                      choices=['onnx', 'onnxruntime', 'None'])
+  args = parser.parse_args()
+
+  cwd_path = Path.cwd()
+  # obtain list of added or modified files in this PR
+  obtain_diff = subprocess.Popen(['git', 'diff', '--name-only', '--diff-filter=AM', 'origin/master', 'HEAD'],
+  cwd=cwd_path, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+  stdoutput, stderroutput = obtain_diff.communicate()
+  diff_list = stdoutput.split()
+
+  # identify list of changed onnx models in model Zoo
+  model_list = [str(model).replace("b'","").replace("'", "") for model in diff_list if ".onnx" in str(model)]
+  # run lfs install before starting the tests
+  test_utils.run_lfs_install()
+
+  print('\n=== Running ONNX Checker on added models ===\n')
+  # run checker on each model
+  failed_models = []
+  tar_ext_name = '.tar.gz'
+  for model_path in model_list:
+      model_name = model_path.split('/')[-1]
+      tar_name = model_name.replace('.onnx', tar_ext_name)
+      print('==============Testing {}=============='.format(model_name))
+
+      try:
+        # replace '.onnx' with '.tar.gz'
+        tar_gz_path = model_path[:-5] + '.tar.gz'
+        test_data_set = []
+        # if tar.gz exists, git pull and try to get test data
+        if os.path.exists(tar_gz_path):
```

Comment:
along with this if you can add one more check for target .i.e 
if os.path.exists(tar_gz_path) and (args.target == 'onnxruntime' or args.target == 'None'):
This way if target is only set to onnx then you dont need to even extract the tar file and then remove again .
If you do this then you dont need to call check_by_target, you can simply call run_backend_ort

Reply:
But I thought the onnx model inside the .tar file needs to be checked by onnx as well? Or we can simply assume it is   same as the .onnx file outside?